packages/hyperion-autologging/src/ALElementValuePublisher.ts

```diff
@@ -0,0 +1,131 @@
+  ALElementEvent &
+  {
+    surface: string;
+    value: string | boolean;
+  }
+>;
+
+export type ALChannelElementValueEvent = Readonly<{
+  al_element_value_event: [ALElementValueEventData],
+}
+>;
+
+export type ALElementValueChannel = Channel<ALChannelElementValueEvent & ALChannelSurfaceMutationEvent>;
+
+
+export type InitOptions = Types.Options<
+  ALSharedInitOptions &
+  {
+    channel: ALElementValueChannel;
+    cacheElementReactInfo: boolean;
+  }
+>;
+
+export function publish(options: InitOptions): void {
+  const { channel } = options;
+
+  channel.addListener('al_surface_mutation_event', surfaceEventData => {
+    const surfaceElement = surfaceEventData.element;
+    const surface = surfaceEventData.surface;
+
+    if (surfaceEventData.event !== 'mount_component' || !surface == null || !surfaceElement) {
+      return;
+    }
+    // The following is expensive, so try to jam everything we are interested in to the selector
+    const elements = surfaceElement.querySelectorAll<HTMLInputElement | HTMLSelectElement>('input[type=radio], input[type=checkbox], select');
+
+    if (!elements.length) {
+      // Nothing to add
+      return;
+    }
+
+    for (let i = 0; i < elements.length; ++i) {
+      const element = elements[i];
+      const parentSurfaceNode = getAncestralSurfaceNode(element);
+      if (parentSurfaceNode === surfaceElement) {
+        // We know value is part of this surface only
+        const elementText = getElementTextEvent(element, surface);
+        let elementValueEventData: Pick<ALElementValueEventData, "value" | "metadata"> | null = null;
+
+        switch (element.nodeName) {
+          case 'INPUT': {
+            const input = element as HTMLInputElement;
+            if (input.checked) {
+              elementValueEventData = {
+                value: input.checked,
+                metadata: {
+                  type: input.getAttribute('type') ?? '',
+                }
+              }
+            }
+            break;
+          }
+          case 'SELECT': {
+            const select = element as HTMLSelectElement;
+            if (select.selectedIndex > -1) {
+              elementValueEventData = {
+                value: select.value,
+                metadata: {
+                  type: 'select',
+                  text: select.options[select.selectedIndex].text
+                }
+              }
+            }
+            break;
+          }
+
+        }
+        if (elementValueEventData) {
+          let reactComponentData: ReactComponentData | null = null;
+          if (options.cacheElementReactInfo) {
+            const elementInfo = ALElementInfo.getOrCreate(element);
+            reactComponentData = elementInfo.getReactComponentData();
+          }
+
+          const flowlet = options.flowletManager.top();
+
+
+          channel.emit('al_element_value_event', {
+            eventIndex: ALEventIndex.getNextEventIndex(),
+            relatedEventIndex: surfaceEventData.eventIndex,
```

Comment:
👍

Reply:
I chose this name for now, wanted to go with `causeEventIndex` but thought this might be too limiting. Thoughts?

Reply:
I think related is good,  and we can describe the scenarios for each case